### PR TITLE
Ported to the latest gensim, line model dimentionality fixed, output format extended

### PR DESCRIPTION
**Pull request by luav**
Fix for #8 (ported to the latest gensim), output extended with the .mat format

### CODE REVIEW

File: src/baseline.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
-from gensim.models import Word2Vec
+from gensim.models.word2vec  import Word2Vec
```
Comment (GTmac) [inline src/baseline.py]: Nit: could you remove that extra space? Thanks!

**Comment by luav** [comment]
>  set the default number of workers to a smaller value (for example, gensim word2vec uses 3

Up to you, in all other functions I set the default number of workers to at least half of the available logical cpus: int(cpu_num() / 2) + 1. Any hardcoded value is not desirable, because the host might have 1 or 2 cores then a harcoded value 3 may affect the performance unlike the value dependent on cpu_num.

> This commit is touching the model part, so have you tested this change in terms of classification

I have not changed any model parameters or internals except adaptation to the updated gensim API. I did perform training and evaluation of the harp + deepwalk / line embeddings on my datasets and they look fine.

**Comment by luav** [comment]
> This seems to be your clone of DeepWalk -- could you change this to the official DeepWalk repo? Thanks!

In the official deepwalk, the walks persistence expects text and not numbers (I made a pull request to the official repository). I'm not sure whether the numerical values necessity there was caused by some some bugs that occurred and fixed during the Harp porting to the updated Deepwalk and gensim, or the numeric walk items are required by Harp from Deepwalk. Anyway, Harp works fine with the extended version (accepting numerical walk items) of Deepwalk in my repository but I have not tested whether it works with the official repository without that extension.

**Comment by luav** [comment]
> This seems to be your clone of DeepWalk -- could you change this to the official DeepWalk repo? Thanks!

In the official deepwalk, the walks persistence expects text and not numbers (I made a [pull request](https://github.com/phanein/deepwalk/issues/98) to the official repository). I'm not sure whether the numerical values necessity there was caused by some some bugs that occurred and fixed during the Harp porting to the updated Deepwalk and gensim, or the numeric walk items are required by Harp from Deepwalk. Anyway, Harp works fine with the extended version (accepting numerical walk items) of Deepwalk in my repository but I have not tested whether it works with the official repository without that extension.

**Comment by luav** [comment]
> > This seems to be your clone of DeepWalk -- could you change this to the official DeepWalk repo? Thanks!
> 
> In the official deepwalk, the walks persistence expects text and not numbers (I made a [pull request](https://github.com/phanein/deepwalk/issues/98) to the official repository). I'm not sure whether the numerical values necessity there was caused by some bugs that occurred and fixed during Harp porting to the updated Deepwalk and gensim, or the numeric walk items are required by Harp from Deepwalk. Anyway, Harp works fine with the extended version (accepting numerical walk items) of Deepwalk in my repository but I have not tested whether it works with the official repository without that extension.

I just verified, the original latest Deepwalk lacks support of the [numerical walk items](https://github.com/phanein/deepwalk/issues/98) to work with HARP, so the specified repository should be used until this [Deepwalk pull request](https://github.com/phanein/deepwalk/pull/99) is merged.

**Comment by luav** [comment]
> > set the default number of workers to a smaller value (for example, gensim word2vec uses 3
> 
> Up to you, in all other functions I set the default number of workers to at least half of the available logical cpus: int(cpu_num() / 2) + 1. Any hardcoded value is not desirable, because the host might have 1 or 2 cores then a harcoded value 3 may affect the performance unlike the value dependent on cpu_num.

Workers number is set to 1 by default and to `cpu_num` for the specified `workers = -1` since
71c490c.

**Pull request closed without merging**